### PR TITLE
WIP: Fix `color` flag behaviour

### LINKED ISSUE #1636: Getting Unicode chars in the report while Newman running even if disableUnicode,noColor options are true.
1. Newman Version (can be found via `newman -v`): **3.8.3 or 3.10.0**
 2. OS details (type, version, and architecture): **OS X EL Capitan, 10.11.6**
 3. Are you using Newman as a library, or via the CLI? LIBRARY
 3. Did you encounter this recently, or has this bug always been there: **Recently after node upgraded**
 4. Expected behaviour: **Unicode chars should be removed**
 5. Command / script used to run Newman: **Script**
 6. Sample collection, and auxiliary files (minus the sensitive details):
 7. Screenshots (if applicable):
screen shot 2018-07-24 at 11 51 40 pm

### PR DESCRIPTION
**Pull request by codenirvana**
**Issue:**
- `noColor|color` options are not handled when using newman as a library.
  - `colors` module internally handle this via `--no-color` flag.
- Bug with `--color` flag
  - This sets `color` option to false.
  - Not affecting previously because of above issue.

**Changes:**
- Explicitly set color if `--color` flag is passed.
- Disable `colors` library's colored output based on `noColor|color` options.
- Add unit tests for `--color` flag.

Fixes #1636

### CODE REVIEW

File: lib/reporters/cli/index.js [hunk 1]
```diff
@@ -68,6 +68,13 @@ PostmanCLIReporter = function (emitter, reporterOptions, options) {
         return; // we simply do not register anything!
     }
 
+    // disable colors based on `noColor|color` option provided when using as a library.
+    // in case of CLI, this is triggered internally by `colors` module via `--no-color` flag.
+    // @todo deprecate noColor option with `color: false`.
```
Comment (kunagpal) [inline lib/reporters/cli/index.js]: It is unlikely that this will ever be done, the comment should be removed.

File: lib/reporters/cli/index.js [hunk 2]
```diff
@@ -68,6 +68,13 @@ PostmanCLIReporter = function (emitter, reporterOptions, options) {
...
+    if (options.noColor === true || options.color === false) {
```
Comment (kunagpal) [inline lib/reporters/cli/index.js]: `options.noColor || !options.color`

File: test/unit/cli.test.js [hunk 3]
```diff
@@ -79,6 +79,41 @@ describe('cli parser', function () {
...
+                    function (err, opts) {
+                        expect(err).to.be.null;
+                        expect(opts.command).to.equal('run');
+                        expect(opts.color, 'should have color to be true').to.equal(true);
+
+                        done();
+                    });
+            });
+
+            it('should have color disabled with --no-color flag', function (done) {
```
Comment (kunagpal) [inline test/unit/cli.test.js]: Also add a test for both flags.

**Comment by shamasis** [comment]
Closing this PR. Pointless to have two commutative options.

@codenirvana the following is what we do

1. Drop support for `--no-color` (ensure it is well documented in migration guide and in API-REF/README with alternative)

2. Add support for parameters to `--color` as `auto`, `off` `on`. Where `--color auto` is default if nothing provided (anything other than off, on and auto is error.)

3. Programmatic input should also have string input `color: "auto"`

**Pull request closed without merging**